enhancements/sig-architecture/29-manifestwork-status-feedback/README.md

```diff
@@ -0,0 +1,253 @@
+
+ 1. A user would like to know the status of the resources applied to the managed cluster, e.g. the user wants to know how many replicas are running in a deployment applied in the ManagedCluster.
+ 2. A higher level controller would require the status to be synced back and collected in ocm. It then can aggregate the collected statuses and return a status summary to the user of the platform.
+ 3. A workload orchestrator would dispatch multiple dependent jobs as a pipeline or DAG across multiple clusters. It needs to get the status of the deployed job on the ocm control plane to decide the next step.
+
+## Motivation
+
+Provide a common approach for users or controllers on the hub cluster to get the status of resources applied by
+`ManifestWork`. A orchestration controller on the hub cluster can use this to design a control loop to distribute
+workloads.
+
+The straightforward approach is to return `status` of arbitrary resources to the hub using `RawExtension`. However, it won't be possible to manage such untyped structure. The size of the whole status field of a resource can be too big to
+be put in the `ManifestWork`. 
+
+Given the current use cases, the controllers or users on the hub only care about certain fields of the applied resources. It may make more sense to explicitly specify the fields of the status in `ManifestWork` spec, and work agent
+will only return the value of these fields.
+
+### Goals
+
+## Proposal
+
+We propose to update `ManifestWork` API so the user can specify the status fields of the resources applied by
+`ManifestWork`. The work-agent will sync these status fields to the `ManifestWork` API.
+
+### Design Details
+
+#### API change  
+
+In the spec of `ManifestWork`, add a new field 
+
+```go
+
+// ManifestWorkSpec represents a desired configuration of manifests to be deployed on the managed cluster.
+type ManifestWorkSpec struct {
+    ...
+	// StatusFeedback defines the option to return status of applied resource
+	StatusFeedbacks []StatusFeedbackOption `json:"statusFeedbacks,omitempty"`
+}
+
+type StatusFeedbackOption struct {
+	// ResourceIdentifier represents the group, resource, name and namespace of a resoure.
+	// It must match to a resource defined in spec.workload.manifests
+	// +required
+	ResourceIdentifier ResourceIdentifier `json:"resourceIdentifier"`
+
+	// FeedBackRules defines the returned resource status feedback
+	FeedbackRule FeedbackRule `json:"feedbackRule"`
+}
+
+type FeedbackRule struct {
+	// Type defines the option of how status can be returned.
+	// It is IntrestedField only today, but can be extend to others in the future.
+	Type string `json:"type"`
+
+	// InterstedFields defines the status fields to be synced for a manifest
+	// An interstedValues will be updated in certain manifest status of the manifestwork.status.
+	InterestedFields []InterestedResourceField `json:"interestedFields,omitempty"`
```

Comment:
In usage, it seems like the same sorts of fields will be requested many times.  Every deployment will likely have an available replica count of something similar.  This can result in subtly different alias names and perhaps even the same alias with different jsonpaths.

While I agree that a design goal of allowing any CustomResource to have a field extracted is valuable, it seems like making the common-case easy and consistent has significant value.  Can add a section to the "alternatives considered" section that shows what an API that allows for some well-known options like the workload status that has `Replicas`, `ReadyReplicas`, and `AvailableReplicas`

Reply:
Added another mode WellKnownStatus so with that use does not need to specify json path. The returned value will have the fixed name.